configure.ac

```diff
@@ -1,225 +1,175 @@
-# wavpack 5.4.0 configure.ac
+dnl wavpack 5.4.0 configure.ac
 
-AC_INIT(wavpack, 5.4.0, bryant@wavpack.com)
+AC_INIT([wavpack], [5.4.0], [bryant@wavpack.com])
 AC_CONFIG_SRCDIR([src/pack.c])
-AM_INIT_AUTOMAKE([serial-tests])
+AM_INIT_AUTOMAKE([-Wall 1.16 serial-tests subdir-objects foreign no-dist-gzip dist-xz])
 AM_MAINTAINER_MODE
 
 LIBWAVPACK_MAJOR=5
 LIBWAVPACK_MINOR=4
 LIBWAVPACK_MICRO=0
-LIBWAVPACK_VERSION_STRING=$LIBWAVPACK_MAJOR.$LIBWAVPACK_MINOR.$LIBWAVPACK_MICRO
-
-AC_DEFINE_UNQUOTED(LIBWAVPACK_MAJOR, $LIBWAVPACK_MAJOR, [libwavpack major version])
-AC_DEFINE_UNQUOTED(LIBWAVPACK_MINOR, $LIBWAVPACK_MINOR, [libwavpack minor version])
-AC_DEFINE_UNQUOTED(LIBWAVPACK_MICRO, $LIBWAVPACK_MICRO, [libwavpack micro version])
-AC_DEFINE_UNQUOTED(LIBWAVPACK_VERSION_STRING, "$LIBWAVPACK_VERSION_STRING", [libwavpack version string])
-
-# set libtool versioning
-# +1 :  0 : +1   == new interface that does not break old one.
-# +1 :  0 :  0   == changed/removed an interface. Breaks old apps.
-#  ? : +1 :  ?   == internal changes that doesn't break anything.
-# CURRENT : REVISION : AGE
+LIBWAVPACK_VERSION_STRING="${LIBWAVPACK_MAJOR}.${LIBWAVPACK_MINOR}.${LIBWAVPACK_MICRO}"
+
+AC_DEFINE_UNQUOTED([LIBWAVPACK_MAJOR], [${LIBWAVPACK_MAJOR}], [libwavpack major version])
+AC_DEFINE_UNQUOTED([LIBWAVPACK_MINOR], [${LIBWAVPACK_MINOR}], [libwavpack minor version])
+AC_DEFINE_UNQUOTED([LIBWAVPACK_MICRO], [${LIBWAVPACK_MICRO}], [libwavpack micro version])
+AC_DEFINE_UNQUOTED([LIBWAVPACK_VERSION_STRING], ["${LIBWAVPACK_VERSION_STRING}"], [libwavpack version string])
+
+dnl set libtool versioning
+dnl +1 :  0 : +1   == new interface that does not break old one.
+dnl +1 :  0 :  0   == changed/removed an interface. Breaks old apps.
+dnl  ? : +1 :  ?   == internal changes that doesn't break anything.
+dnl CURRENT : REVISION : AGE
 LT_CURRENT=3
 LT_REVISION=3
 LT_AGE=2
 
-AC_SUBST(LT_CURRENT)
-AC_SUBST(LT_REVISION)
-AC_SUBST(LT_AGE)
+AC_SUBST([LT_CURRENT])
+AC_SUBST([LT_REVISION])
+AC_SUBST([LT_AGE])
 
-# Check for os version
-VERSION_OS=$(uname -s)
-AC_DEFINE_UNQUOTED(VERSION_OS, "$VERSION_OS", [os version])
-
-# Check host so we can infer target CPU (and assembly optimizations)
+dnl Check for os version
 AC_CANONICAL_HOST
+AC_DEFINE_UNQUOTED([VERSION_OS], ["${host_os}"], [os version])
 
-# Check for processor characteristics
-AC_C_BIGENDIAN(AC_DEFINE([HIGHFIRST], [1], [big-endian machine]))
```

Comment:
I checked the whole codebase and found no mention of either `HIGHFIRST` or `WORDS_BIGENDIAN`, so seems like dead code?

Reply:
The code does check for `BYTE_ORDER` or `__BYTE_ORDER__` for endianess. It's less than optimum if those are not defined (slower code), but we will only generate bad code if they're defined and wrong (which doesn't seem likely). And yes, those you mentioned are not used anywhere.

Reply:
You're right, but neither `AC_C_BIGENDIAN` nor its "success" block define `BYTE_ORDER` or `__BYTE_ORDER__` (both of which aren't portable), hence why I removed the macro.

Reply:
I'm happy to do whatever's the correct method, but I do need to know whether I'm building for a big-endian or little-endian machine unless I want to get a significant performance hit. Is there a portable way to do that?

Reply:
`AC_C_BIGENDIAN` is the correct way for that, you just aren't using its results to affect compilation. What about replacing `BYTE_ORDER` and friends with `WORDS_BIGENDIAN`, which is defined by `AC_C_BIGENDIAN` on big-endian machines? I can test this on our sparc machines.

Reply:
I actually think we're okay now that I've looked a little closer. The macros I check for are defined by the compiler (which is probably the most credible source) and I check for whether they're defined before I use them (so we can't error out). The purpose is to _only_ detect little endian, which I have some optimizations for, but the code will still work fine even if it's not detected (and the performance difference is not that great). No need for `AC_C_BIGENDIAN`.

Reply:
(as an aside, the code around the endianness dispatch includes undefined behavior: you're casting an `unsigned char*` to a `uint16_t*`, which violates strict aliasing rules, and with newer compiler versions/LTO/-O3/something might break. In general, pointer casting is a dangerous practice and should be avoided.)